ibm/resource_ibm_dns_reverse_record.go

```diff
@@ -0,0 +1,149 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+func resourceIBMDNSReverseRecord() *schema.Resource {
+	return &schema.Resource{
+		Exists:   resourceIBMDNSREVERSERecordExists,
+		Create:   resourceIBMDNSREVERSERecordCreate,
+		Read:     resourceIBMDNSREVERSERecordRead,
+		Update:   resourceIBMDNSREVERSERecordUpdate,
+		Delete:   resourceIBMDNSREVERSERecordDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			"ipaddress": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"hostname": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ttl": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				DefaultFunc: func() (interface{}, error) {
+					return 604800, nil
+				},
+			},
+		},
+	}
+}
+
+//  Creates DNS Domain Reverse Record
+//  https://sldn.softlayer.com/reference/services/SoftLayer_Dns_Domain/CreatePtrRecord
+func resourceIBMDNSREVERSERecordCreate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	service := services.GetDnsDomainService(sess.SetRetries(0))
+	Data := sl.String(d.Get("hostname").(string))
+	Ttl := sl.Int(d.Get("ttl").(int))
+	Ipaddress := sl.String(d.Get("ipaddress").(string))
+	var id int
+	record, err := service.CreatePtrRecord(Ipaddress, Data, Ttl)
+	if record.Id != nil {
+		id = *record.Id
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error creating DNS Reverse %s", err)
+	}
+	d.SetId(fmt.Sprintf("%d", id))
+	log.Printf("[INFO] Dns Reverse %s ", d.Id())
+	return resourceIBMDNSREVERSERecordRead(d, meta)
+}
+
+//  Reads DNS Domain Reverse Record from SL system
+//  https://sldn.softlayer.com/reference/services/SoftLayer_Dns_Domain_ResourceRecord/getObject
+func resourceIBMDNSREVERSERecordRead(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	service := services.GetDnsDomainResourceRecordService(sess)
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid ID, must be an integer: %s", err)
+	}
+	hostname := sl.String(d.Get("hostname").(string))
```

Comment:
d.Set should set the values from result object line 80 ..Please don't use d.Get() in read method.
And remove log.print which are not important